build/toolchain/gcc_toolchain.gni

```diff
@@ -185,9 +185,21 @@ template("gcc_toolchain") {
     }
 
     tool("link") {
-      outfile = "{{root_out_dir}}/{{target_output_name}}{{output_extension}}"
+      exename = "{{target_output_name}}{{output_extension}}"
+      outfile = "{{root_out_dir}}/$exename"
       rspfile = "$outfile.rsp"
-      command = "$ld {{ldflags}} -o $outfile -Wl,--start-group @$rspfile {{solibs}} -Wl,--end-group $libs_section_prefix {{libs}} $libs_section_postfix"
+      unstripped_outfile = outfile
+
+      if (defined(invoker.strip)) {
```

Comment:
add a section to the top-of-file comment documenting what the "strip" variable does.  in particular, it's useful to document exactly which files end up where when this is set or not set.  you can crib the one from chromium